ttnn/cpp/ttnn/tensor/shape.cpp

```diff
@@ -13,7 +13,7 @@ bool SimpleShape::operator==(const std::vector<uint32_t> &other) const {
 }
 
 size_t SimpleShape::rank() const {
-    return this->m_value.size();
+    return this->m_original_size;
```

Comment:
lets create a method `size` in VectorBase and reuse it here. Seems like leakage of internal implementation details of base.

Reply:
This member is protected for a reason. I'd like to avoid having method `size` in the base

Reply:
Fixed